test/gtest/kernel_tuning_net.cpp

```diff
@@ -46,11 +46,16 @@ struct KernelTuningNetTest : public ::testing::TestWithParam<KernelTuningNetTest
             tensor<G>(test_case.data_type, test_case.layout, test_case.weight);
         tensor<G> output_tensor =
             tensor<G>(test_case.data_type, test_case.layout, test_case.output);
-        problem        = miopen::ProblemDescription(input_tensor.desc,
+
+        const bool is_forward = (test_case.direction == miopen::conv::Direction::Forward);
+        const auto conv_problem =
+            miopen::conv::ProblemDescription(is_forward ? input_tensor.desc : output_tensor.desc,
                                              weight_tensor.desc,
-                                             output_tensor.desc,
+                                             is_forward ? output_tensor.desc : input_tensor.desc,
```

Comment:
[Q] Is this change temporary?

Reply:
No, that's how it's meant to be. Is there something wrong?

Reply:
I do not know? probably not. But why this looks ugly?